kdump/include/runtest.sh

```diff
@@ -511,7 +506,86 @@ AppendConfig()
         shift
     done
 
-    rstrnt-report-log -l "${KDUMP_CONFIG}"
+    RstrntSubmit "${KDUMP_CONFIG}"
+}
+
+
+AppendSysconfig()
+{
+    Log "- Modifying /etc/sysconfig/kdump"
+
+    local KEY=$1
+    local ACTION=$2
+    local VALUE1=$3
+    local VALUE2=$4
+
+    if [ -z "$KEY" ] || [ -z "$ACTION" ] || [ -z "$VALUE1" ]; then
+        Error "- Missing KEY or ACTION or VALUE1."
+        return 1
+    elif [ "$ACTION" = "replace" ] && [ -z "$VALUE2" ]; then
+        Error "- Missing new_value for replacing."
+        return 1
+    elif ! grep "^$KEY=\"" "${KDUMP_SYS_CONFIG}"; then
+        Error "- Invalid KEY: $KEY."
+        return 1
+    fi
+
+    local kdump_sys_config_tmp="${KDUMP_SYS_CONFIG}.tmp"
+    \cp "${KDUMP_SYS_CONFIG}" "${kdump_sys_config_tmp}"
+
+    # Note: When KEY is "KDUMP_COMMANDLINE", ACTION add/remove/replace is actually
+    # manipulating the value of current kernel cmdline and assign
+    # it to KDUMP_COMMANDLINE.
+    # if there is no value set to KDUMP_COMMANDLINE, assign current kernel cmdline
+    # to it for later string manipluation.
+    if [ "$KEY" == "KDUMP_COMMANDLINE" ] && \
+        grep "^$KEY=[\ \"]*$" "${kdump_sys_config_tmp}"; then
+
+        sed -i "/^KDUMP_COMMANDLINE=\"/d" "${kdump_sys_config_tmp}"
+        echo "KDUMP_COMMANDLINE=\"$(cat /proc/cmdline)\"" >> "${kdump_sys_config_tmp}"
+    fi
+
+    case $ACTION in
+        add)
+            # Check if the key already has a value on it, then append " $VALUE1" to the
+            # original value.
+            Log "- Add '$VALUE1' to '$KEY'"
+            if grep -q "^$KEY=[\ \"]*$" "${kdump_sys_config_tmp}"; then
+                sed -i /^"$KEY="/d "${kdump_sys_config_tmp}"
+                echo "$KEY=\"$VALUE1\"" >> "${kdump_sys_config_tmp}"
+            else
+                sed -i "/^$KEY=/s/\"/\"$VALUE1 /" "${kdump_sys_config_tmp}"
+            fi
+            ;;
+        remove)
+            Log "- Remove '$VALUE1' from '$KEY'"
+            sed -i  "/^$KEY=/s/$VALUE1//g" "${kdump_sys_config_tmp}"
+            ;;
+        replace)
+            Log "- Replace '$VALUE1' with '$VALUE2' for '$KEY'"
+            sed -i  "/^$KEY=/s/$VALUE1/$VALUE2/g" "${kdump_sys_config_tmp}"
+            ;;
+        override)
+            Log "- Set '$VALUE1' to '$KEY'"
+            sed -i "/^$KEY=\"/d" "${kdump_sys_config_tmp}"
+            echo "$KEY=\"$VALUE1\"" >> "${kdump_sys_config_tmp}"
+            ;;
+        *)
+            Error "Invalid action '${ACTION}' for editing kdump sysconfig."
+            false
+            ;;
+    esac
+
+    [ $? -ne 0 ] && {
+        Error "- Failed to edit kdump sysconfig"
+        rm -f "${kdump_sys_config_tmp}"
+        return 1
+    }
+
+    \mv "${kdump_sys_config_tmp}" "${KDUMP_SYS_CONFIG}"
+    RstrntSubmit "${KDUMP_SYS_CONFIG}"
+    sync;sync;sync
```

Comment:
Now that I think of it, maybe add a comment with the explanation why three `sync` calls?

Reply:
Updated. Added a comment next to 3 sync calls.